src/Data/Rank1Dynamic.hs

```diff
@@ -101,6 +102,15 @@ instance Show Dynamic where
 toDynamic :: Typeable a => a -> Dynamic
 toDynamic x = Dynamic (typeOf x) (unsafeCoerce x)
 
+-- | Construct a dynamic value with a user-supplied type rep
+--
+-- This function is unsafe because we have no way of verifying that the
+-- provided type representation matches the value.
+--
+-- Since 0.3.3.2.
```

Comment:
The latest hackage version is 0.3.1.1. I think you can count on the next one to be 0.3.2.0.